planning/mission_planner/src/mission_planner/mission_planner.cpp

```diff
@@ -273,19 +302,112 @@ void MissionPlanner::on_set_mrm_route(
+  }
+
+  // Use temporary pose stamped for transform.
+  PoseStamped pose;
+
+  // Convert route points.
+  PlannerPlugin::RoutePoints points;
+  points.push_back(odometry_->pose.pose);
+  for (const auto & waypoint : req->waypoints) {
+    pose.pose = waypoint;
+    points.push_back(transform_pose(pose).pose);
+  }
+  pose.pose = req->goal;
+  points.push_back(transform_pose(pose).pose);
+
+  // Plan route.
+  LaneletRoute new_route = planner_->plan(points);
+  if (new_route.segments.empty()) {
+    throw component_interface_utils::ServiceException(
+      ResponseCode::ERROR_PLANNER_FAILED, "The planned route is empty.");
+  }
+  new_route.header.stamp = req->header.stamp;
+  new_route.header.frame_id = map_frame_;
+  new_route.uuid.uuid = generate_random_id();
+  new_route.allow_modification = req->option.allow_goal_modification;
+
+  // check route safety
+  if (checkRerouteSafety(*normal_route_, new_route)) {
+    // sucess to reroute
+    change_mrm_route(new_route);
+    change_state(RouteState::Message::SET);
+    res->status.success = true;
+  } else {
+    // failed to reroute
+    change_route(*normal_route_);
+    change_state(RouteState::Message::SET);
+    res->status.success = false;
+  }
 }
 
 // NOTE: The route interface should be mutually exclusive by callback group.
 void MissionPlanner::on_clear_mrm_route(
-  const ClearMrmRoute::Service::Request::SharedPtr req,
+  [[maybe_unused]] const ClearMrmRoute::Service::Request::SharedPtr req,
   const ClearMrmRoute::Service::Response::SharedPtr res)
 {
-  // TODO(Yutaka Shimizu): reroute for MRM
-  (void)req;
-  (void)res;
+  using ResponseCode = autoware_adapi_v1_msgs::srv::SetRoutePoints::Response;
+
+  if (!planner_->ready()) {
+    throw component_interface_utils::ServiceException(
+      ResponseCode::ERROR_PLANNER_UNREADY, "The planner is not ready.");
+  }
+  if (!odometry_) {
+    throw component_interface_utils::ServiceException(
+      ResponseCode::ERROR_PLANNER_UNREADY, "The vehicle pose is not received.");
+  }
+  if (!normal_route_) {
+    throw component_interface_utils::ServiceException(
+      ResponseCode::ERROR_PLANNER_UNREADY, "Normal route is not set.");
+  }
+  if (!mrm_route_) {
+    throw component_interface_utils::ServiceException(
+      ResponseCode::ERROR_PLANNER_UNREADY, "Mrm route is not set.");
+  }
+
+  // check route safety
+  if (checkRerouteSafety(*mrm_route_, *normal_route_)) {
+    clear_mrm_route();
+    change_route(*normal_route_);
+    res->success = true;
+    return;
+  }
+
+  // reroute with normal goal
+  PlannerPlugin::RoutePoints points;
+  points.push_back(odometry_->pose.pose);
+  points.push_back(normal_route_->goal_pose);
+
+  // Plan route.
+  LaneletRoute new_route = planner_->plan(points);
+  if (new_route.segments.empty()) {
+    RCLCPP_ERROR_THROTTLE(get_logger(), *get_clock(), 5000, "Reroute with normal goal failed.");
+    res->success = false;
+    return;
+  }
```

Comment:
I think it also is better to check `checkRerouteSafety(*mrm_route_, new_route)`.
if it is `false` return

Reply:
and change_route(*mrm_route_)

Reply:
done https://github.com/autowarefoundation/autoware.universe/pull/3466/commits/dc175782d2141c241ee9061b23f81ee75de3f66c